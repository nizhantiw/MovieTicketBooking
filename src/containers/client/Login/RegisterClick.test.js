import RegisterClick from './RegisterClick';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import { render, fireEvent } from "@testing-library/react";
import renderer from 'react-test-renderer'
//import { render } from 'react-dom';
configure({adapter: new Adapter()});

it("renders correctly", () => {
  const wrapper = shallow(
    <RegisterClick />
  );
  expect(wrapper).toMatchSnapshot();
});

it('should show the text', () => {
  const toggleInstance = shallow(<RegisterClick />);
  const element = toggleInstance.find('div h3');
  expect(element.text()).toMatchSnapshot();
 });

 it("show the sended message", () => {
  const onSubmit = jest.fn();
  const { getByLabelText, getByText } = render(<RegisterClick />);
  const inputValue = "abc";
  const inputValue2 = "abc";
  const inputValue3 = "abc";
  //const inputValue4 = "abc";

  fireEvent.change(getByLabelText("UserId"), { target: { value: inputValue } });
  fireEvent.change(getByLabelText("EmailId"), { target: { value: inputValue2 } });
  fireEvent.change(getByLabelText("Password"), { target: { value: inputValue3 } });
 // fireEvent.change(getByLabelText("ConfirmPassword"), { target: { value: inputValue4 } });
  fireEvent.click(getByText(/Submit/i));


  //expect(onSubmit).toBeCalled();
  expect(onSubmit).toHaveBeenCalledTimes(0);
});

//

it('renders a snapshot', () => {
    const tree = renderer.create(<RegisterClick/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  


   it('renders results after search', () => {
    const wrapper = shallow(<RegisterClick/>);
    expect(wrapper.find('h3').length).toEqual(1);
  });
  it('starts with a count of 0', () => {
      const wrapper = shallow(<RegisterClick />);
      const text = wrapper.find('h3').text();
      expect(text).toEqual('Sigup with your details');
    });

/*
      it('should render correctly', () => {
        expect(render).toHaveBeenCalledWith(
          <RegisterClick/>, 'element-node'
        );
        expect(render).toHaveBeenCalledTimes(1);
      });
*/