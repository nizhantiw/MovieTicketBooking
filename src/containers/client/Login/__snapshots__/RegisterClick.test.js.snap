// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div
  style={
    Object {
      "backgroundColor": "lightcyan",
      "height": "100vh",
      "textAlign": "center",
    }
  }
>
  <br />
  <h3>
    Sigup with your details
  </h3>
  <br />
  <form
    class="submit"
    onSubmit={[Function]}
  >
    <div>
      <label
        htmlFor="UserId"
      >
        UserId
      </label>
      <input
        id="UserId"
        name="bankName"
        onChange={[Function]}
        placeholder="UserId"
        required={true}
        type="text"
        value=""
      />
    </div>
    <br />
    <div>
      <label
        htmlFor="EmailId"
      >
        EmailId
      </label>
      <input
        id="EmailId"
        name="accNumber"
        onChange={[Function]}
        placeholder="EmailId"
        required={true}
        type="text"
        value=""
      />
    </div>
    <br />
    <div>
      <label
        htmlFor="Password"
      >
        Password
      </label>
      <input
        id="Password"
        name="IFSC"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="IFSC"
        value=""
      />
    </div>
    <br />
    <div>
      <label
        htmlFor="ConfirmPassword"
      >
        Confirm Password
      </label>
      <input
        id="ConfirmPassword"
        name="CVV"
        onChange={[Function]}
        placeholder="Confirm Password"
        required={true}
        type="CVV"
        value=""
      />
    </div>
    <br />
    <input
      type="submit"
      value="Submit"
    />
  </form>
</div>
`;

exports[`renders correctly 1`] = `ShallowWrapper {}`;

exports[`should show the text 1`] = `"Sigup with your details"`;
